api/src/main/java/com/graphhopper/client/AbstractHopperClient.java

```diff
@@ -0,0 +1,122 @@
+        routeResponse.setDistance(Helper.round(response.getDistance(), 3));
+        routeResponse.setWeight(Helper.round6(response.getDistance()));
+        routeResponse.setTime(response.getMillis());
+
+        BBox bbList = response.calcRouteBBox(new BBox(0, 0, 0, 0));
+        RouteBbox bbox = new RouteBbox();
+        bbox.setMaxLatitude(bbList.maxLat);
+        bbox.setMinLatitude(bbList.minLat);
+        bbox.setMaxLongitude(bbList.maxLon);
+        bbox.setMinLongitude(bbList.minLon);
+        routeResponse.setBbox(bbox);
+
+        PointList pointList = response.getPoints();
+
+        List<double[]> coordinates = getCoordinates(pointList);
+        routeResponse.setPolyline(WebHelper.encodePolyline(pointList));
+        routeResponse.setCoordinates(coordinates);
+        routeResponse.setPoints(new LazyPointList(coordinates));
+
+
+        // Paths.Instructions[]
+        List<RouteInstruction> routeInstructionList = new ArrayList<RouteInstruction>(response.getInstructions().size());
+        InstructionList responseInstructions = response.getInstructions();
+        if (responseInstructions != null && !responseInstructions.isEmpty()) {
+            List<Map<String, Object>> createJson = responseInstructions.createJson();
+            for (int i = 0; i < createJson.size(); i++) {
+                Map<String, Object> jsonInstruction = createJson.get(i);
+                RouteInstruction instruction = new RouteInstruction();
+                instruction.setDistance((Double) jsonInstruction.get("distance"));
+                instruction.setSign((Integer) jsonInstruction.get("sign"));
+                instruction.setTime((Long) jsonInstruction.get("time"));
+                instruction.setText((String) jsonInstruction.get("text"));
+
+                //noinspection unchecked
+                List<Integer> interval = (List<Integer>) jsonInstruction.get("interval");
+                instruction.setInterval(new int[]{interval.get(0), interval.get(1)});
+
+                if (jsonInstruction.containsKey("annotation_text") && jsonInstruction.containsKey("annotation_importance")) {
+                    instruction.setAnnotationText((String) jsonInstruction.get("annotation_text"));
+                    instruction.setAnnotationImportance((Integer) jsonInstruction.get("annotation_importance"));
+                }
+
+                PointList pointList1 = response.getInstructions().get(i).getPoints();
+
+                instruction.setPolyline(WebHelper.encodePolyline(pointList1));
+                instruction.setPoints(new LazyPointList(coordinates.subList(interval.get(0), interval.get(1))));
+                instruction.setCoordinates(coordinates.subList(interval.get(0), interval.get(1)));
+                routeInstructionList.add(instruction);
+            }
+        }
+
+        routeResponse.setInstructions(routeInstructionList);
+
+        return routeResponse;
+    }
+
+    private List<double[]> getCoordinates(PointList pointList) {
```

Comment:
Why not use pointList.getLatitude(index) instead of toGeoJson? Also please avoid being ironic, not really helpful to understand what you mean.